dex/farm-staking-contracts/metabonding-staking/src/locked_asset_token.rs

```diff
@@ -0,0 +1,109 @@
+elrond_wasm::imports!();
+elrond_wasm::derive_imports!();
+
+use factory::locked_asset_token_merge::ProxyTrait as _;
+
+pub type PaymentsVec<M> = ManagedVec<M, EsdtTokenPayment<M>>;
+
+#[derive(TypeAbi, TopEncode, TopDecode, Debug, PartialEq)]
+pub struct StakingEntry<M: ManagedTypeApi> {
+    pub nonce: u64,
+    pub amount: BigUint<M>,
+    pub opt_unbond_epoch: Option<u64>,
+}
+
+impl<M: ManagedTypeApi> StakingEntry<M> {
+    pub fn new(nonce: u64, amount: BigUint<M>) -> Self {
+        Self {
+            nonce,
+            amount,
+            opt_unbond_epoch: None,
+        }
+    }
+
+    #[inline]
+    pub fn is_unstaked(&self) -> bool {
+        self.opt_unbond_epoch.is_some()
+    }
+}
+
+#[elrond_wasm::module]
+pub trait LockedAssetTokenModule {
+    fn require_all_locked_asset_payments(&self, payments: &PaymentsVec<Self::Api>) {
+        require!(payments.len() > 0, "No payments");
+
+        let locked_asset_token_id = self.locked_asset_token_id().get();
+        for p in payments {
+            require!(
+                p.token_identifier == locked_asset_token_id,
+                "Invalid payment"
+            );
+        }
+    }
+
+    fn merge_locked_asset_tokens_if_needed(
+        &self,
+        user_address: &ManagedAddress,
+        mut new_tokens: PaymentsVec<Self::Api>,
+    ) -> EsdtTokenPayment<Self::Api> {
+        let entry_mapper = self.staking_entry_for_user(user_address);
```

Comment:
shouldn't this function also clear this storage?

Reply:
Nope. It's being overwritten in the caller function:
```
self.staking_entry_for_user(&caller).set(&StakingEntry::new(
            new_locked_asset_token.token_nonce,
            new_locked_asset_token.amount,
        ));
```